### PR TITLE
[FileUpload] Allow to encode custom scalar types as null and use it for file uploads

### LINKED ISSUE #2409: graphql-multipart-request-spec is not followed fully
**Summary**

Hello, I am facing this issue. Please, check it first:
https://github.com/graphql-java-kickstart/graphql-java-servlet/issues/260

**Version**
apollo-api-jvm:2.2.1

**Description**

I am getting the following exception:
```
java.lang.RuntimeException: expected null value when mapping variables.avatar
	at graphql.kickstart.servlet.core.internal.VariableMapper.mapVariable(VariableMapper.java:51) ~[graphql-java-servlet-9.2.0.jar:na]
	at graphql.kickstart.servlet.GraphQLMultipartInvocationInputParser.lambda$null$8(GraphQLMultipartInvocationInputParser.java:109) ~[graphql-java-servlet-9.2.0.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1510) ~[na:na]
	at graphql.kickstart.servlet.GraphQLMultipartInvocationInputParser.lambda$mapMultipartVariables$9(GraphQLMultipartInvocationInputParser.java:109) ~[graphql-java-servlet-9.2.0.jar:na]
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:723) ~[na:na]
	at graphql.kickstart.servlet.GraphQLMultipartInvocationInputParser.mapMultipartVariables(GraphQLMultipartInvocationInputParser.java:103) ~[graphql-java-servlet-9.2.0.jar:na]
	at graphql.kickstart.servlet.GraphQLMultipartInvocationInputParser.lambda$getGraphQLInvocationInput$2(GraphQLMultipartInvocationInputParser.java:77) ~[graphql-java-servlet-9.2.0.jar:na]
	at java.base/java.util.Optional.ifPresent(Optional.java:176) ~[na:na]
	at graphql.kickstart.servlet.GraphQLMultipartInvocationInputParser.getGraphQLInvocationInput(GraphQLMultipartInvocationInputParser.java:77) ~[graphql-java-servlet-9.2.0.jar:na]
```

This happens because during file upload you set the value to be equal `mime` type instead of setting it to be null like it is done in the spec: https://github.com/jaydenseric/graphql-multipart-request-spec. 

I expect you to make it configurable or nullable. Also, this can be fixed at `graphql-java-kickstart/graphql-java-servlet` side, but technically, its you don't follow this spec, — not them: they expect it to be null as it must be. So, IMO, this should be fixed on your side first.

Thanks

### PR DESCRIPTION
**Pull request by martinbonnin**
fixes #2409

### CODE REVIEW

**Comment by martinbonnin** [comment]
Ping @meiblorn. Sorry for the delay getting this in, let me know if this looks good for you.

**Comment by meiblorn** [comment]
@martinbonnin, thanks a lot. I haven't tested it yet, but overall it looks good.

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/json/InputFieldJsonWriter.kt [hunk 1]
```diff
@@ -80,6 +80,7 @@ class InputFieldJsonWriter(
       is GraphQLString -> writeString(fieldName, customTypeValue.value)
       is GraphQLBoolean -> writeBoolean(fieldName, customTypeValue.value)
       is GraphQLNumber -> writeNumber(fieldName, customTypeValue.value)
+      is GraphQLNull -> writeString(fieldName, null)
```
Comment (sav007) [inline apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/json/InputFieldJsonWriter.kt:83]: nit: should we introduce `writeNull` to our `JsonWriter`? 😄

Comment (martinbonnin) [reply apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/json/InputFieldJsonWriter.kt:83]: `JsonWriter` already has `nullValue`, would that be ok? 

We could also add `writeNull` to `InputFieldWriter` but that would be a much more invasive change that I'm not too keen on doing in 2.x. Falling into the rabbit hole of `InputFieldsWriters`, I realized that `SortedInputFieldMapWriter` doesn't handle custom scalar types. So custom scalars as field arguments will most likely fill the cache with different values. I guess something to keep in mind for `3.x`

**Review by sav007 — APPROVED** [review]

**Pull request merged**